src/ci/docker/dist-x86_64-musl/Dockerfile

```diff
@@ -41,6 +40,13 @@ ENV RUST_CONFIGURE_ARGS \
 # See: https://github.com/rust-lang/rust/issues/34978
 ENV CFLAGS_x86_64_unknown_linux_musl=-Wa,-mrelax-relocations=no
 
-ENV SCRIPT \
-      python2.7 ../x.py test --target x86_64-unknown-linux-musl && \
-      python2.7 ../x.py dist --target x86_64-unknown-linux-musl
+ENV HOSTS=x86_64-unknown-linux-musl \
+    CC_x86_64_unknown_linux_musl=x86_64-linux-musl-gcc \
+    CXX_x86_64_unknown_linux_musl=x86_64-linux-musl-g++
+
+#    CARGO_TARGET_ARM_UNKNOWN_LINUX_MUSLEABIHF_LINKER=musl-gcc \
+#    CARGO_TARGET_ARM_UNKNOWN_LINUX_MUSLEABIHF_RUNNER="qemu-arm -L /musl-arm"
+
+ENV RUSTFLAGS="-C target-feature=-crt-static"
```

Comment:
Could this be passed through to just the musl target via `config.toml` perhaps? Ideally we wouldn't set this flag for compilations like of bootstrap itself or the original glibc compiler

Reply:
Should be doable, I didn't use `crt-static` from `config.toml` because it will make the toolchain ]link dynamically by default](https://github.com/rust-lang/rust/blob/master/config.toml.example#L471).

Reply:
Hm I'm incorrectly remembering that `config.toml` has an option to specify `rustflags`-per-target. Since it doesn't actually have that this request can be safely ignored!

Reply:
I'm just now seeing this. Sorry the description for that option isn't quite clear. Building a `musl`-host `rustc` is exactly what it's for: it only enables building of `dylib`s for `libstd`, and affects how `rustc` *gets linked*. It does *not* affect the behavior of the final binary.

Reply:
For what it's worth, at Exherbo for native musl hosts, we also disable crt-static. We *do* want rustc to be dynamically linked to the system musl

Reply:
Thanks @smaeul I'll try that.